### DISCUSSION TITLE
JobRunr in Cloud Run from GCP

### DISCUSSION THREAD

**Discussion by ro0t** [jobrunr/jobrunr#839] [Q&A]
Hello there,

I saw there was another question regarding AppEngine on GCP but I'm using Cloud Run to host my Spring Boot application.

Has anyone managed to get JobRunr working with a Cloud Run server? Would be appreciated to get some insights on how the setup should be, if it's possible to run it on Cloud run that is.

**Comment by rdehuyss** [comment]
From what I understand from Google Cloud Run is that these are normal docker images with normal Java threads.

If that is the case, I don't see why JobRunr would not work.

**Comment by ro0t** [comment]
Yeah that was my thought as well but it's not launching the JobRunr service when it's on Cloud Run, no logs either.

I'll try to figure out why and will post back here if I find the solution 🤠 Probably some configuration thing.

**Comment by ro0t** [comment]
Been digging into this and can't get it working properly. It registers jobs (in the SQL tables) on the cloud instance but the background servers are not firing up/doing anything on the cloud instance. I have to spin up the project locally to process the jobs, lol. 

CloudRun only allows of exporting one port, so I can't get both Spring Boot and the Dashboard running on the same container. I also never see any JobRunr logs when running on Cloud Run, which is pretty weird even if everything is enabled via application.properties or even trying to initiate via Fluent API. 

Tried bundling a few services into one Nginx Docker container that runs Spring Boot + Nginx to proxy to the :8000 port but the background servers are just not being launched in the logs. Any ideas why?
